plugins/http2_adapter/test/http2_test.dart

```diff
@@ -76,4 +76,45 @@ void main() {
     final res = await dio.post('post', data: 'TEST');
     expect(res.data.toString(), contains('TEST'));
   });
+
+  test('catch DioException when receiveTimeout', () {
+    final dio = Dio(
+      BaseOptions(
+        baseUrl: 'https://httpbun.com/',
+        receiveTimeout: Duration(seconds: 5),
+      ),
+    );
+    dio.httpClientAdapter = Http2Adapter(
+      ConnectionManager(
+        idleTimeout: Duration(milliseconds: 10),
+      ),
+    );
+
+    expectLater(
+      dio.get('/drip?delay=10&numbytes=1'),
+      allOf([
+        throwsA(isA<DioException>()),
+        throwsA(predicate(
+            (DioException e) => e.type == DioExceptionType.receiveTimeout)),
+        throwsA(predicate(
+            (DioException e) => e.message!.contains('0:00:05.000000'))),
+      ]),
+    );
+  }, testOn: 'vm');
```

Comment:
Since the adapter only supports on VM, should we use `@TestOn('vm')` for the file instead?